### DISCUSSION TITLE
How to get api-key

### DISCUSSION THREAD

**Discussion by vlaterz** [datalust/seq-tickets#1593] [Help] [answered]
Greetings!

I need to create a ps1 installer with SEQ included, so that user will not have to configure anything (without UI, including SEQ Web ui).
SEQ service shall be listening to localhost.
 
After service installation i got stuck with creating and retreiving api-key.
I need to create it and write it into config file, which will be used by my App. 

_**I didn`t manage to figure out, how to create api-key via console, using seq.exe and seqcli.exe.**_ 

I tried to set firstRun.adminUsername and firstRun.adminPasswordHash with "seq auth reset -u admin -p password", but SEQ requires to change this password after service installation (you should use web UI to change it, as i understood). So that, i cannot create any api-key with "seqcli apikey create" with my own pre-set admin login/password passed as parameters. It just returns : "401 - A password change is required." 

I also tried to create a user ( seqcli user create -n user -d 'user example' -r 'User (read/write)' -p password --no-password-change), but seq returned: "401 - Please log in."

How can i change pre defined admin login and password, so that SEQ will not prompt for password change?
Or how can create user with seq.exe?

**Comment by nblumhardt** [comment, marked as answer]
Hi @vlaterz - when `firstRun.adminUsername` and `firstRun.adminPasswordHash` are specified, no password change should be required by default.

You can then create an API key with:

```
$user = "admin"
$pw = "thepassword"
$token = (
  echo $pw |
  seqcli apikey create `
    -t CLI `
    --permissions="read,write,project,organization,system" `
    --connect-username $user --connect-password-stdin
)
```

You'll only be able to use the "firstRun" settings before Seq has ever started up, though, so that might be where things are getting stuck. HTH!

**Reply by vlaterz** [reply]
Well, i didn`t succeed. =)

What i did:
seq service stop
seq service uninstall
seq config set -k firstRun.adminUsername=admin
seq config hash -> (typed in 'password') -> got [hash]
seq config set -k firstRun.adminPasswordHash=[hash]
service install
service restart

After that i tried to log in on web ui (login:admin / password:password) It said:
![image](https://user-images.githubusercontent.com/33147329/175489405-1e9070be-c17f-46d2-baab-f99f744f9871.png)

Also tried logging in with hash as password (from seq config) - failed again.

I also tried to set adminPasswordHash as 'password' (not as hash, just string)
Result is the same

seqcli apikey create returns the same in all variants. What am i doing wrong?

**Reply by vlaterz** [reply]
@nblumhardt

**Reply by nblumhardt** [reply]
Hi @vlaterz;

My guess currently is that after starting fresh, there's still an initialized `Documents/` folder in Seq's storage location, so the "first run" configuration won't be applied. You need to delete all data from disk before each attempt at setting "first run" options.

If you have a supported license and need more help working through it, `support@datalust.co` is the way to go; it's not always possible to loop back to threads here in a very short time as we're balancing quite a few things going on concurrently :-)

Just to call out also, in case it saves you some wasted effort or headaches down the track - Seq's EULA doesn't support redistribution (including of the Individual version), which may be problematic if your users are outside your organization - please drop us a line if you need more info.

Best regards,
Nick
